File: CSharp/Example1/ExampleClass.cs
namespace Example1
{
    public class ExampleClass : IExample
    {
        public string Name { get; set; }

        // Constructor por defecto - sin parámetros, inicializa con valores por defecto
        public ExampleClass()
        {
            Name = "Default";
        }

        // Implementación de interfaz - cuando la interfaz es implementada
        public void PerformAction()
        {
            Console.WriteLine("Accion realizada en Example1");
        }
    }
}

File: Java/Example1/ExampleClass.java
package Example1;

public class ExampleClass implements IExample {
    private String name;

    // Constructor por defecto - sin parámetros, inicializa con valores por defecto
    public ExampleClass() {
        this.name = "Default";
    }

    // Implementación de interfaz - cuando la interfaz es implementada
    @Override
    public void performAction() {
        System.out.println("Accion realizada en Example1");
    }
}